activation/validation.go

```diff
@@ -114,7 +121,7 @@ func (v *Validator) NIPost(
 }
 
 func (v *Validator) PoetMembership(
-	ctx context.Context,
+	_ context.Context,
```

Comment:
if context is ignored, do we actually need it on the method signature?

Reply:
It should be removed then. I wonder if there is a linter in golangci-lint that could catch unused function arguments?

Reply:
There is: https://golangci-lint.run/usage/linters/#unparam. It reports 125 unused function parameters across the repo.

Reply:
I can remove this one, but I would suggest removing others in separate PR with some other refactoring, if it is needed

Reply:
Yeah linter changes should be their own PR, they can be quite big and would just add noise to this one.